lib/matplotlib/figure.py

```diff
@@ -3061,10 +3061,24 @@ def savefig(self, fname, *, transparent=None, **kwargs):
             if transparent:
                 kwargs.setdefault('facecolor', 'none')
                 kwargs.setdefault('edgecolor', 'none')
+                # set subfigure to appear transparent in printed image
+                for subfig in self.subfigs:
```

Comment:
Subfigures can be nested arbitrarily; this only handles one level.

Reply:
I think another issue was supposed to be opened to handle this general case.

Reply:
The nesting should be handled here.

Reply:
@tacaswell noted how to do this recursively here https://github.com/matplotlib/matplotlib/pull/22816#issuecomment-1101845649
This could also be done with a stack approach:
```
subfigs = [*self.subfigs]
while subfigs:
    this_subfig = subfigs.pop(0)
    subfigs.extend(this_subfig.subfigs)
    # Do something with this_subfig
```
On a side note (and not for this PR), I wonder if we want some kind of `self.iter_subfigures` or similar that would do the above iteration with a callable.